datastructures/set_test.go

```diff
@@ -0,0 +1,37 @@
+package datastructures
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func generateNewSetWithData() *Set[int] {
+	set := MakeSet[int]()
+	set.Add(3)
+	set.Add(5)
+	set.Add(7)
+
+	return set
+}
+
+func TestSetExistsAndAdd(t *testing.T) {
+	testSet := generateNewSetWithData()
+	assert.True(t, true, testSet.Exists(3))
+	assert.False(t, false, testSet.Exists(7))
```

Comment:
Please notice that the second parameter of this assertion is the expected value. The third one is optional.
```suggestion
	assert.True(t, testSet.Exists(3))
	assert.False(t, testSet.Exists(4))
}
```